apps/wallet/src/ui/app/pages/home/transfer-coin/SendTokenForm.tsx

```diff
@@ -0,0 +1,334 @@
+}: SendTokenFormProps) {
+    const aggregateBalances = useAppSelector(accountAggregateBalancesSelector);
+    const coinBalance = useMemo(
+        () => (coinType && aggregateBalances[coinType]) || BigInt(0),
+        [coinType, aggregateBalances]
+    );
+
+    const allCoins = useAppSelector(accountCoinsSelector);
+    const coins = useMemo(
+        () =>
+            allCoins.filter(
+                ({ type }) => CoinAPI.getCoinType(type) === coinType
+            ),
+        [allCoins, coinType]
+    );
+
+    const suiCoins = useMemo(
+        () =>
+            allCoins.filter(
+                ({ type }) => CoinAPI.getCoinType(type) === SUI_TYPE_ARG
+            ),
+        [allCoins]
+    );
+
+    const gasAggregateBalance = aggregateBalances[SUI_TYPE_ARG] || BigInt(0);
+
+    const coinSymbol = (coinType && CoinAPI.getCoinSymbol(coinType)) || '';
+
+    const [coinDecimals] = useCoinDecimals(coinType);
+    const [gasDecimals] = useCoinDecimals(SUI_TYPE_ARG);
+
+    const maxSuiSingleCoinBalance = useIndividualCoinMaxBalance(SUI_TYPE_ARG);
+
+    const validationSchemaStepOne = useMemo(
+        () =>
+            createValidationSchemaStepOne(
+                coinType || '',
+                coinBalance,
+                coinSymbol,
+                gasAggregateBalance,
+                coinDecimals,
+                gasDecimals,
+                0,
+                maxSuiSingleCoinBalance
+            ),
+        [
+            coinType,
+            coinBalance,
+            coinSymbol,
+            coinDecimals,
+            gasDecimals,
+            gasAggregateBalance,
+            maxSuiSingleCoinBalance,
+        ]
+    );
+
+    const [tokenBalance, symbol, queryResult] = useFormatCoin(
+        coinBalance,
+        coinType,
+        CoinFormat.FULL
+    );
+
+    // Max token depending on coin response could be '...' base on the query result
+    // remove the comma from the token balance
+    const maxToken =
+        tokenBalance !== '...' ? tokenBalance.replace(/,/g, '') : null;
+
+    return (
+        <Loading loading={queryResult.isLoading}>
+            <Formik
+                initialValues={{
+                    amount: initialAmount,
+                    to: initialTo,
+                    isPayAllSui:
+                        initialAmount === maxToken && coinType === SUI_TYPE_ARG,
+                    gasInputBudgetEst: 0,
+                }}
+                validationSchema={validationSchemaStepOne}
+                enableReinitialize
+                validateOnMount
+                validateOnChange
+                onSubmit={({
+                    to,
+                    amount,
+                    isPayAllSui,
+                    gasInputBudgetEst,
+                }: FormValues) => {
+                    if (!gasInputBudgetEst || !coins || !suiCoins) return;
+                    const coinsIDs = coins
+                        .sort((a, b) => a.fields.balance - b.fields.balance)
```

Comment:
Is this sorting coins in ascending order? I thought we sort so we make sure the first contains enough balance to pay for gas. Is this still the reason we sort here?